src/plots/mapbox/mapbox.js

```diff
@@ -367,7 +367,10 @@ proto.resolveOnRender = function(resolve) {
         if(map.loaded()) {
             map.off('render', onRender);
             // resolve at end of render loop
-            setTimeout(resolve, 0);
+            //
+            // Need a 10ms delay (0ms should suffice to skip a thread in the
+            // render loop) to workaround mapbox-gl bug introduced in v1.3.0
+            setTimeout(resolve, 10);
```

Comment:
@archmoj w/o this extra 10ms delay, call sequences like

```js
// off the `mapbox_0` mock
Plotly.restyle(gd, 'visible', false)
.then(() => Plotly.restyle(gd, 'visible', true))
.then(() => Plotly.restyle(gd, 'visible', 'legendonly', [1]))
```

fails and logs

![image](https://user-images.githubusercontent.com/6675409/65774683-c4da2a80-e10c-11e9-9e73-4f651e9c182b.png)

--------------

I suspect this is caused by https://github.com/mapbox/mapbox-gl-js/pull/8633 - but I haven't investigated further.

Reply:
OK. I'll leave this conversation open so that it is noticed by in the QA.